src/System.IO.Compression.ZipFile/tests/ZipFileInvalidFileTests.cs

```diff
@@ -46,55 +46,85 @@ public void InvalidConstructors()
         [Fact]
         public void InvalidFiles()
         {
-            ConstructorThrows<InvalidDataException>(() => ZipFile.OpenRead(bad("EOCDmissing.zip")));
-            ConstructorThrows<InvalidDataException>(() => ZipFile.Open(bad("EOCDmissing.zip"), ZipArchiveMode.Update));
-            ConstructorThrows<InvalidDataException>(() => ZipFile.OpenRead(bad("CDoffsetOutOfBounds.zip")));
-            ConstructorThrows<InvalidDataException>(() => ZipFile.Open(bad("CDoffsetOutOfBounds.zip"), ZipArchiveMode.Update));
+            using (TempFile testArchive = CreateTempCopyFile(bad("EOCDmissing.zip"), GetTestFilePath()))
+            {
+                ConstructorThrows<InvalidDataException>(() => ZipFile.OpenRead(testArchive.Path));
```

Comment:
You don't need it where Zip.Open/ZipFile.OpenRead is used, as it already disposes the FileStream if something fails: https://github.com/dotnet/corefx/blob/master/src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.cs#L205

But e.g. in this line https://github.com/dotnet/corefx/blob/master/src/System.IO.Compression.ZipFile/tests/ZipFileInvalidFileTests.cs#L67 where ZipArchive constructor is directly called I think you need it as the constructor calls the Init function which opens multiple streams which need to be disposed in case of an error. Right?